### PR TITLE
Remove `@Homebrew/plc` from `CODEOWNERS`

### PR DESCRIPTION
**Pull request by Rylan12**
Follow-up from https://github.com/Homebrew/homebrew-core/pull/126267

`@Homebrew/plc` can't be listed in `CODEOWNERS` because it doesn't have write access to this repository. I've removed it from the file, but to avoid a situation where only one person is listed for a given file, I've added `@Homebrew/tsc` as owners for `LICENSE.txt` and `audit_exceptions/permitted_formula_license_mismatches.json`. Even though licensing definitely falls under PLC and not TSC for these things, I think it's better to have this than no owners at all (or just a single owner).

I'm open to other thoughts, though

### CODE REVIEW

**Review by carlocab — APPROVED** [review]

**Review by carlocab — DISMISSED** [review]
Suggestion: add the PLC members who do currently have write access to the lines that would've had `@Homebrew/plc`.

**Comment by Rylan12** [comment]
As an additional thought: the PL will always be a part of `@Homebrew/tsc` (at least with the current governance rules). Should we remove the explicit @MikeMcQuaid reference?

Mike: if you want to keep the explicit one that's fine with me. I only suggest this since it will mean that ownership will seamlessly transfer if there's a new PL elected, and you wouldn't be review-requested anymore if you don't want to be

**Review by carlocab — DISMISSED** [review]

**Review by issyl0 — DISMISSED** [review]
Works for me. Not that I know much about licenses so I will be deferring to PLC/TSC folks who do, but that doesn't require them to have write access.

File: CODEOWNERS
Comment (Rylan12) [inline CODEOWNERS:21]: Originally I just replaced `plc` with `tsc` here, but after adding Issy and Sean, should I remove the `tsc` requirement?

```suggestion
LICENSE.txt                       @MikeMcQuaid @issyl0 @SMillerDev
```

Comment (SMillerDev) [inline CODEOWNERS:21]: I don't think members should be explicitly named. I prefer to leave it up to the PL to alert the PLC if needed

Comment (MikeMcQuaid) [reply CODEOWNERS:21]:
```suggestion
LICENSE.txt                       @Homebrew/tsc @MikeMcQuaid
```

agreed. let's make it like this.

File: CODEOWNERS
Comment (MikeMcQuaid) [inline CODEOWNERS:14]:
```suggestion
#
# Note: @Homebrew/plc does not have write-access to this repository, and therefore cannot be listed in this file.
```

**Review by MikeMcQuaid — APPROVED** [review]

**Comment by MikeMcQuaid** [comment]
Thanks @Rylan12 and @SMillerDev!

**Pull request merged**
